include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,555 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides seqan3::views::minimiser.
+ */
+
+#pragma once
+
+#include <deque>
+
+#include <seqan3/range/hash.hpp>
+#include <seqan3/range/views/detail.hpp>
+
+namespace seqan3::detail
+{
+// ---------------------------------------------------------------------------------------------------------------------
+// minimiser class
+// ---------------------------------------------------------------------------------------------------------------------
+
+/*!\brief The type returned by seqan3::views::minimiser.
+ * \tparam urng_t The type of the underlying ranges, must model std::forward_range, the reference type must model
+ *                std::integral. The typical use case is that the reference type is the result of seqan3::kmer_hash.
+ * \implements std::ranges::view
+ * \implements std::ranges::random_access_range
+ * \implements std::ranges::sized_range
+ * \ingroup views
+ *
+ * \details
+ *
+ * Note that most members of this class are generated by ranges::view_interface which is not yet documented here.
+ */
+template <std::ranges::view urng_t>
+class minimiser : public std::ranges::view_interface<minimiser<urng_t>>
+{
+private:
+    static_assert(std::ranges::forward_range<urng_t const>, "The minimiser only works on forward_ranges.");
+    static_assert(std::is_integral<reference_t<urng_t>>::value, "The reference type of the underlying range must model "
+                                                                "std::integral.");
+    //!\brief The underlying range.
+    urng_t urange;
+
+    //!\brief The number of elements in one window.
+    uint32_t num_w_elems;
```

Comment:
It is not the window_size anymore bur the number of elements in a window, so I think naming it window_size would be confusing.

Reply:
I thought the two are synonyms.
Doesn't a window size of 10 mean 10 elements in the window?

Reply:
No, a window_size of 10 means w - k + 1 elements that are completely in the window, we can also say we want to consider all k-mers starting in a window (which is done in some papers), but in all our example figures we do the former... But this view should be capable of handling both, it depends on what an user defines for num_w_elems, so I still would say not naming it window_size is a good idea.

Reply:
ah ok. thanks for the explanation. I would still try to avoid abbreviations as it makes it easier to read the code without jumping to the documentation of the variable.
What about `elements_in_window` or `window_values_size` (if I remember the name of your std::deque variable that stores the window values/elements). You also switch between the name "elements" or "values", it would be easier to understand if you'd only use one of them for consistency?

Reply:
Ok, I changed the variable name to `window_values_size` and removed all mentions of the name element in the context of the minimiser.